### PR TITLE
Create rust workflow

### PR DESCRIPTION
**Pull request by maryamtahhan**
Add a build to run the rust pipeline

### CODE REVIEW

**Comment by KeithWiles** [comment]
Missing a DCO :-)

**Comment by maryamtahhan** [comment]
> Missing a DCO :-)

Yep I will fixup when I get the pipeline running :) sorry about that.

the pipeline is failing unexpectedly with: 

```
The following warnings were emitted during compilation:

warning: src/c_src/bindings.c:5:10: fatal error: pktdev.h: No such file or directory
warning:     5 | #include <pktdev.h>          // for pktdev_rx_burst, pktdev_tx_burst
warning:       |          ^~~~~~~~~~
warning: compilation terminated.

error: failed to run custom build command for `cndp-cne-sys v0.2.0 (https://github.com/CloudNativeDataPlane/cndp?branch=main#f86286d5)`

Caused by:
  process didn't exit successfully: `/home/runner/work/cndp/cndp/lang/rs/target/debug/build/cndp-cne-sys-978e8b1bf782987d/build-script-build` (exit status: 1)
  --- stdout
  ```

**Comment by maryamtahhan** [comment]
> Must be missing a step.
TY, Yeah I think so... just trying to update the pipeline now based on the README :)

File: .github/workflows/rust-build.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - name: update rust
+      run: rustup update stable
+
+    - name: Build
+      working-directory: lang/rs
+      run: cargo build
+
+    - name: Run Tests
+      working-directory: lang/rs/apis/cne
+      run: ./run_cne_test.sh
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply .github/workflows/rust-build.yml:42]: Looks like it is missing a newline at the end of the file.

**Comment by KeithWiles** [comment]
:-( have a great night.

**Comment by maryamtahhan** [comment]
I think there's a bit more to do with the test debug for now - so I'm putting it on the back burner with an [issue](https://github.com/CloudNativeDataPlane/cndp/issues/291) to look at later... that way we can still get this workflow for at least building the rust code running regularly.

**Comment by maryamtahhan** [comment]
yes - this is ready.

The unit tests need to be fixed separately and have issues open against them - so they are out of scope for this PR

**Review by KeithWiles — APPROVED** [review]

**Pull request merged**